### PR TITLE
Removed synchronized services from Symfony 2.7 docs

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7+ |
| Fixed tickets | #5140 |

### CODE REVIEW

File: cookbook/service_container/scopes.rst [hunk 1]
```diff
@@ -96,169 +97,28 @@ when compiling the container. Read the sidebar below for more details.
...
-        ->addMethodCall('setClientConfiguration', array(
-            new Reference(
-                'client_configuration',
-                ContainerInterface::NULL_ON_INVALID_REFERENCE,
-                false
-            )
-        ));
+    In Symfony 2.6 and previous versions, there was another alternative based
+    on ``synchronized`` services. However, these kind of services have been
+    removed starting from Symfony 2.7.
```
Comment (cordoval) [inline cookbook/service_container/scopes.rst]: Perhaps can we say why, or do some background even if it is a sentence? Or point to the upgrade2.7.md bullet that explains this? I know this feature is not very popular but i am finding that at least mentioning a migration path is a good thing.

Comment (WouterJ) [inline cookbook/service_container/scopes.rst]: It isn't removed, it's deprecated.

And I'm against making this note unnessecary longer.

File: cookbook/service_container/scopes.rst [hunk 2]
```diff
@@ -96,169 +97,28 @@ when compiling the container. Read the sidebar below for more details.
...
-            new Definition('AppBundle\Mail\Mailer')
-        )
-        ->addMethodCall('setClientConfiguration', array(
-            new Reference(
-                'client_configuration',
-                ContainerInterface::NULL_ON_INVALID_REFERENCE,
-                false
-            )
-        ));
+    In Symfony 2.6 and previous versions, there was another alternative based
```
Comment (WouterJ) [inline cookbook/service_container/scopes.rst]: "Prior to Symfony 2.7, [...]"

**Comment by stof** [comment]
@javiereguiluz the next question is whether we should actually document the usage of custom scopes (IIRC, we discussed removing the request scope in 3.0 as the RequestContext is the proper way to handle this, and then the question is whether we actually need scopes)

File: cookbook/service_container/scopes.rst (hunk 2 above)
Comment (javiereguiluz) [inline cookbook/service_container/scopes.rst]: Much better! Thanks.

**Comment by WouterJ** [comment]
@stof I propose to wait until a decision has been made about the future of scopes.

File: cookbook/service_container/scopes.rst [hunk 3]
```diff
@@ -96,169 +97,28 @@ when compiling the container. Read the sidebar below for more details.
...
-        ->addMethodCall('setClientConfiguration', array(
-            new Reference(
-                'client_configuration',
-                ContainerInterface::NULL_ON_INVALID_REFERENCE,
-                false
-            )
-        ));
+    Prior to Symfony 2.7, there was another alternative based on ``synchronized``
+    services. However, these kind of services have been deprecated starting from
+    Symfony 2.7.
```
Comment (xabbuh) [inline cookbook/service_container/scopes.rst]: Should we add the old label (`using-synchronized-service`) before this note? So when people follow the old URL, they will get this explanation in the future.

Comment (javiereguiluz) [inline cookbook/service_container/scopes.rst]: I've just added the `.. _using-synchronized-service:` label. Thanks

**Comment by xabbuh** [comment]
:+1:

**Comment by weaverryan** [comment]
Great job Javier!

**Pull request merged**